### PR TITLE
create_apis_for_station

### PR DESCRIPTION
**Pull request by hau-tran**
Following this issue: https://github.com/mgm-interns/team-radio/issues/277

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,115 +1,115 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (hau-tran) [inline pom.xml]: e đang tìm cách sửa file ni, tự nhiên nó bị cách lồi ra, a review giúp e mấy file ở dưới nhé ^^

File: src/main/java/com/mgmtp/radio/controller/v1/StationController.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            notes = "Create station"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Request processed successfully", response = RadioSuccessResponse.class),
+            @ApiResponse(code = 400, message = "Error in request parameters", response = RadioBadRequestException.class),
+            @ApiResponse(code = 500, message = "Server error", response = RadioException.class)
+    })
+    @PostMapping
+    public Mono<ResponseEntity<StationDTO>> createStation(@Valid @RequestBody StationDTO stationDTO) {
+        return stationService.create("string", stationDTO)
```
Comment (hau-tran) [inline src/main/java/com/mgmtp/radio/controller/v1/StationController.java]: vì e chưa đăng nhập được nên e làm user giả

File: src/main/java/com/mgmtp/radio/controller/v1/StationController.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            value = "GET all station",
+            notes = "Returns all station"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Request processed successfully", response = RadioSuccessResponse.class),
+            @ApiResponse(code = 500, message = "Server error", response = RadioException.class)
+    })
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    public Flux<ResponseEntity<StationDTO>> getAllStation() {
```
Comment (hau-tran) [inline src/main/java/com/mgmtp/radio/controller/v1/StationController.java]: có cần xác định chỉ tìm public stations k anh?

File: src/main/java/com/mgmtp/radio/mapper/decorator/SongMapperDecorator.java [hunk 4]
```diff
@@ -15,8 +17,13 @@
...
-		songDTO.setUpVoteCount(song.getUpVoteUserIdList().size());
-		songDTO.setDownVoteCount(song.getDownVoteUserIdList().size());
+		songDTO.setUpVoteCount(getVoteCount(song.getUpVoteUserIdList()));
+		songDTO.setDownVoteCount(getVoteCount(song.getDownVoteUserIdList()));
 		return songDTO;
 	}
-}
+
+	private int getVoteCount(List<String> voteList) {
+		if (voteList == null) return 0;
```
Comment (dat-ng) [inline src/main/java/com/mgmtp/radio/mapper/decorator/SongMapperDecorator.java]: return (voteList == null ? 0 : voteList.size());

File: src/main/java/com/mgmtp/radio/service/station/SongServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    public SongServiceImpl(SongMapper songMapper, SongRepository songRepository) {
+        this.songMapper = songMapper;
+        this.songRepository = songRepository;
+    }
+
+    @Override
+    public Flux<SongDTO> getAllSongById(List<String> idList) {
+        return songRepository.findAllById(idList)
+                .map( song -> songMapper.songToSongDTO(song) ).defaultIfEmpty(new SongDTO());
```
Comment (dat-ng) [inline src/main/java/com/mgmtp/radio/service/station/SongServiceImpl.java]: .map(songMapper::songToSongDTO).defaultIfEmpty(SongDTO::new);

File: src/main/java/com/mgmtp/radio/service/station/StationServiceImpl.java [hunk 6]
```diff
@@ -1,7 +1,63 @@
...
+                .map(station -> stationMapper.stationToStationDTO(station));
+    }
+
+    @Override
+    public Mono<StationDTO> findById(String id) {
+        Mono<Station> stationMono = stationRepository.findById(id);
+        Station station = stationMono.block();
+        StationDTO stationDTO = stationMono.map(stationMapper::stationToStationDTO).block();
+        List<String> playlistIdList = station.getPlaylist();
+        if (playlistIdList == null || playlistIdList.isEmpty()) return Mono.just(stationDTO);
```
Comment (dat-ng) [inline src/main/java/com/mgmtp/radio/service/station/StationServiceImpl.java]: should have bracket {}

File: src/main/java/com/mgmtp/radio/service/station/StationServiceImpl.java [hunk 7]
```diff
@@ -1,7 +1,63 @@
...
+    @Override
+    public Mono<StationDTO> create(String userId, StationDTO stationDTO){
+        stationDTO.setOwnerId(userId);
+        stationDTO.setCreatedAt(LocalDate.now());
+        Station station = stationMapper.stationDtoToStation(stationDTO);
+        return stationRepository.save(station).map( item -> stationMapper.stationToStationDTO(item) );
+    }
+
+    @Override
+    public Mono<StationDTO>  update(String stationId, StationDTO stationDTO){
```
Comment (dat-ng) [inline src/main/java/com/mgmtp/radio/service/station/StationServiceImpl.java]: space before "update"

**Pull request merged**